src/Mod/Part/BOPTools/ToleranceFeatures.py

```diff
@@ -108,19 +109,26 @@ def __init__(self,obj):
         obj.addProperty("App::PropertyLinkList","Objects","ToleranceSet","Objects to have tolerance adjusted.")
         obj.addProperty("App::PropertyBool","Refine","ToleranceSet",
                         "True = refine resulting shape. False = output as is.")
-        obj.addProperty("App::PropertyFloat","minTolerance","ToleranceSet", "1e-7")
+        obj.addProperty("App::PropertyLength","minTolerance","ToleranceSet", "0.1 nm")
+        obj.addProperty("App::PropertyLength","maxTolerance","ToleranceSet", "0")
         obj.Refine = getParamRefine()
 
         obj.Proxy = self
         self.Type = "FeatureToleranceSet"
 
+    def onChanged(self, obj, prop):
```

Comment:
It's a somewhat ugly way to extend old FeatureToleranceSet with the new property. Therefore you can implement the method: `def onDocumentRestored(self, obj):`

This way the check is done only once and not every time a property has changed.

Reply:
thanks, great suggestion, I didn't know that.